pkg/logs/input/file/file_provider.go

```diff
@@ -130,6 +131,20 @@ func (p *Provider) searchFiles(pattern string, source *config.LogSource) ([]*Fil
 		return nil, fmt.Errorf("could not find any file matching pattern %s, check that all its subdirectories are executable", pattern)
 	}
 	var files []*File
+
+	// Files are sorted because of a heuristic on the filename: often the filename and/or the folder name
+	// contains information in the file datetime. Most of the time we want the most recent files.
+	// Here, we reverse paths to have stable sort keep reverse lexicographical order w.r.t dir names. Example:
+	// [/tmp/1/2017.log, /tmp/1/2018.log, /tmp/2/2018.log] becomes [/tmp/2/2018.log, /tmp/1/2018.log, /tmp/1/2017.log]
+	// then kept as is by the sort below.
+	for i := len(paths)/2 - 1; i >= 0; i-- {
+		opp := len(paths) - 1 - i
+		paths[i], paths[opp] = paths[opp], paths[i]
+	}
+	// sort paths by descending filenames
+	sort.SliceStable(paths, func(i, j int) bool {
+		return filepath.Base(paths[i]) > filepath.Base(paths[j])
```

Comment:
Isn't that the same as paths[j] > paths[i] without having to reverse paths before ?

Reply:
No because here we sort only filename whereas the reverse ensure the whole filepath is reversed. This covers 2 distinct usecases:

```bash
├── abc123
│   ├── 1.log
│   └── 2.log
└── def456
    ├── 1.log
    └── 2.log
```

where we want the `2.log` files to come out first and 

```bash
├── 2018-01-17
│   └── error.log
└── 2018-01-18
    └── error.log
```

There are some edge case that are not well handled though, mainly 

```bash
├── 2018-01-17
│   ├── access.log
│   └── error.log  # <- this one is tailed before 
└── 2018-01-18
    ├── access.log # <- this one
    └── error.log
```

or 

```bash
├── abc123
│   ├── 1.log
│   ├── 2.log
│   ├── 3.log  # <-┐
│   ├── 4.log  # <-└ these are not active but tailed before
│   └── 5.log  
└── def456
    ├── 1.log
    └── 2.log  # <- this one which is active in a k8s context
```

It's hard (impossible?) to find a good solution for this without that is also generic